pkg/compactor/block_upload_test.go

```diff
@@ -1178,42 +1167,33 @@ func TestMultitenantCompactor_HandleBlockUpload_Complete(t *testing.T) {
 			expInternalServerError: true,
 		},
 		{
-			name:     "uploading meta file fails",
+			name:     "uploading validation file fails",
 			tenantID: tenantID,
 			blockID:  blockID,
 			setUpBucketMock: func(bkt *bucket.ClientMock) {
-				bkt.MockExists(metaPath, false, nil)
 				metaJSON, err := json.Marshal(validMeta)
 				require.NoError(t, err)
+				bkt.MockExists(metaPath, false, nil)
 				setUpGet(bkt, uploadingMetaPath, metaJSON, nil)
 				setUpGet(bkt, validationPath, nil, bucket.ErrObjectDoesNotExist)
-				bkt.MockUpload(metaPath, fmt.Errorf("test"))
+
+				bkt.MockUpload(validationPath, fmt.Errorf("fail"))
 			},
 			expInternalServerError: true,
 		},
 		{
-			name:     "removing in-flight meta file fails",
```

Comment:
Note: I wish we could keep this test case, since it's useful.

Reply:
This and the other previously deleted tests have now been revived. It ended up being a bit trickier than I initially thought. I did it by converting the mock usage to an `InMemBucket` paired with adding an internal configuration to have the validation be synchronous. This avoids the test having to poll or leak goroutines. 

A nice consequence of this is that some of the subtests now have to complete the entire validation (which is currently still spare).